rewrite-java/src/main/java/org/openrewrite/java/cleanup/UseJavaInterface.java

```diff
@@ -0,0 +1,214 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.java.cleanup;
+
+import org.openrewrite.ExecutionContext;
+import org.openrewrite.Recipe;
+import org.openrewrite.Tree;
+import org.openrewrite.internal.ListUtils;
+import org.openrewrite.java.JavaIsoVisitor;
+import org.openrewrite.java.JavaVisitor;
+import org.openrewrite.java.search.UsesType;
+import org.openrewrite.java.tree.*;
+import org.openrewrite.marker.Markers;
+
+import java.time.Duration;
+import java.util.*;
+
+public class UseJavaInterface extends Recipe {
+
+    @Override
+    public String getDisplayName() {
+        return "Use `Deque`, `List`, `Map`, `ConcurrentMap`, `Queue`, `Set`, `SortedMap`, and `SortedSet`";
+    }
+
+    @Override
+    public String getDescription() {
+        return "Replaces declarations of implemented classes in Java collection with the equivalent interface.";
+    }
+
+    @Override
+    public Set<String> getTags() {
+        return Collections.singleton("RSPEC-1319");
+    }
+
+    @Override
+    public Duration getEstimatedEffortPerOccurrence() {
+        return Duration.ofMinutes(10);
+    }
+
+    @Override
+    protected JavaVisitor<ExecutionContext> getSingleSourceApplicableTest() {
+        List<String> rspecTargetTypes = Arrays.asList(
+                "java.util.ArrayDeque",
+                "java.util.concurrent.ConcurrentLinkedDeque",
+                "java.util.AbstractList",
+                "java.util.AbstractSequentialList",
+                "java.util.ArrayList",
+                "java.util.LinkedList",
+                "java.util.concurrent.CopyOnWriteArrayList",
+                "java.util.AbstractMap",
+                "java.util.EnumMap",
+                "java.util.HashMap",
+                "java.util.Hashtable",
+                "java.util.IdentityHashMap",
+                "java.util.LinkedHashMap",
+                "java.util.WeakHashMap",
+                "java.util.concurrent.ConcurrentHashMap",
+                "java.util.concurrent.ConcurrentSkipListMap",
+                "java.util.AbstractQueue",
+                "java.util.concurrent.ConcurrentLinkedQueue",
+                "java.util.concurrent.SynchronousQueue",
+                "java.util.AbstractSet",
+                "java.util.EnumSet",
+                "java.util.HashSet",
+                "java.util.LinkedHashSet",
+                "java.util.concurrent.CopyOnWriteArraySet",
+                "java.util.TreeMap",
+                "java.util.TreeSet");
+        return new JavaIsoVisitor<ExecutionContext>() {
+            @Override
+            public J.CompilationUnit visitCompilationUnit(J.CompilationUnit cu, ExecutionContext executionContext) {
+                for (String fqn : rspecTargetTypes) {
+                    doAfterVisit(new UsesType<>(fqn));
+                }
```

Comment:
This feels a bit inefficient. Consider making `rspecTargetTypes` a `Set` and iterating over `cu.getTypesInUse()` checking each type for membership in your the `rspecTargetTypes` set.

Reply:
Even better, use `TypeUtils.isAssignableTo(..)`. Then you don't need to list these at all.

Reply:
An RSpec Set sounds reasonable, I’ll add that when I can pick this up again.

Tyler mentioned isAssignableTo as well, but that would catch more than what is listed. Guava, etc.

Also, the rules have specific replacements while the targets implement multiple types.